src/main/java/ucb/internship/backend/controllers/InternshipController.java

```diff
@@ -0,0 +1,61 @@
+package ucb.internship.backend.controllers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ucb.internship.backend.dtos.InternshipListDto;
+import ucb.internship.backend.dtos.ResponseDto;
+import ucb.internship.backend.services.impl.InternshipServiceImpl;
+import ucb.internship.backend.dtos.InternshipDto;
+import ucb.internship.backend.models.Internship;
+
+import java.sql.Date;
+import java.util.List;
+
+@RestController
+@CrossOrigin(origins = "*")
+public class InternshipController {
+    private InternshipServiceImpl internshipService;
+    public static final Logger LOGGER = LoggerFactory.getLogger(InternshipController.class);
+    @Autowired
+    public InternshipController(InternshipServiceImpl internshipService) {
+        this.internshipService = internshipService;
+    }
+    @PostMapping("/internship")
+    public ResponseEntity<String> createInternship(@RequestBody InternshipDto internshipDto){
+        LOGGER.info("Creating internship {}", internshipDto);
+        String response = internshipService.createInternship(internshipDto);
+        if(response.equals("Error creating internship")){
+            return new ResponseEntity<>(response, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return new ResponseEntity<>(response, HttpStatus.CREATED);
+    }
+    @GetMapping("/internship/{id}")
+    public List<Internship> getInternship(@PathVariable Integer id){
+        return internshipService.getInternshipById(id);
+    }
+    @GetMapping("/internship")
+    public ResponseEntity<ResponseDto<Page<InternshipListDto>>> getInternship(
+            @RequestParam(required = false) String city,
+            @RequestParam(required = false) String major,
+            @RequestParam(required = false) Date startingDate,
+            @RequestParam(required = false) Date endingDate,
+            @RequestParam(required = false) Integer page,
+            @RequestParam(required = false) Integer size){
```

Comment:
You don't need to add all the params like this, you can create a record class that holds this instead